azure-kusto-ingest/azure/kusto/ingest/_resource_manager.py

```diff
@@ -131,3 +134,10 @@ def get_ingestions_status_tables(self) -> List[_ResourceUri]:
     def get_authorization_context(self):
         self._refresh_authorization_context()
         return self._authorization_context
+
+    def retrieve_service_type(self):
+        command_result = self._kusto_client.execute("NetDefaultDB", _SHOW_VERSION)
```

Comment:
Re-opening the issue,
The user cares not that you decided to implicitly run _.show version_
Should execute fail, he would not know what to do with the server exception.
You can choose to,